doc/specs/#120,121,119-Update-Uninstall-List.md

```diff
@@ -0,0 +1,230 @@
+| **-s,--source** |   Find the application using the specified [source](source.md). |
+| **-e,--exact**     | Find the application using exact match. |
+ 
+
+By default **List** will show those apps installed through the Windows Package Manager.  However developers should be able to view all apps currently installed via other means.  
+
+
+
+**--all** 
+All will allow the Windows Package Manager to provide **list** data on all apps whether installed through the Package Manager or not.
+
+To obtain this data, the Windows Package Manager will query the 
+**Apps and Features**  installed apps.  This appears to be a superset of 
+**Add Remove Programs** (ARP).
+
+
+### Installed with Windows Package Manager
+If the app is installed with the Windows Package Manager, we have the ID to map the installed version to the repository version.  If the version installed on the PC is lower than the the version in the repository, the list command will show the available version in the "available update."
+
+### Installed out of band from the Windows Package Manaager but in repo
+If the app is installed out of band from the Package Manager, the app will be discovered if it registers correctly with the **Apps and Features**.  The Windows Package Manager will map the installed app back to the repo.  If the version installed is less than the version available in the repo, the **list** command will show the available version in the "available update."
+
+If the installed version is greater than or equal to the version of the app available in the repo, then the "available update" will be left blank.
+
+### Installed out of band from the Windows Package Manaager but not in repo
+If the app is installed out of band from the Package Manager, and is not available in the repo, then the Windows Package Manager will treat it as if there are no updates available.
+
+
+
+
+## Mapping between Windows Package Manager repo and Apps and Features
+*StoreApps*  
+Store apps will be mapped via the The Package Family Name (which is composed by Publisher and Name).   The Package Family Name will be consistent between Apps and Features and the Windows Package Manager repo
+
+*MSIs and EXEs*  
+Will be mapped to the Windows Package Manager repo via the SystemAppId.  The SystemAppId is a required field.  The value in the field differs depending on the InstallerType.
+
+For MSI it is the product code.  Typically a GUID that is typically found in the uninstall registry location and includes the brackets.
+For example: ```{5740BD44-B58D-321A-AFC0-6D3D4556DD6C}```
+
+```HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{3740BD44-B58D-321A-AFC0-6D3D4556DD6C}]```
+    
+```HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\{3740BD44-B58D-321A-AFC0-6D3D4556DD6C}]```
+
+For inno, wix, nullsoft, and exe, the SystemAppId should be a string that is located in either of the Uninstall keys above. 
+
+### Upgrade
```

Comment:
It would be nice to hear how syncing with the remote repository will be handled, will the user have to do this manually (i.e. using an `update`), or will it be done automatically if the repository hasn't been synced after a said time?

Reply:
Martin, I am curious.  I **think** that the syncing of the remote repository is handled via the source:
C:\Users\kevinla>winget source update
Updating all sources...
Updating source: winget...
  ██████████████████████████████  100%
Done.

Do you think this addresses the Update concern?

Reply:
It is true you can handle this using the `winget source` command. I just think the spec should outline how the syncing of the remote repo is handled in context of running an update. Is a user required to run an `winget source update` before running `winget upgrade` or will this be implicit from running `winget upgrade`? (i.e. upgrade will invoke source update)

Reply:
In Linux package managers update the package cache for all enabled repositories. If a package was installed from repository "foo" then the search for available updates for that specific application will be narrowed down to repository "foo". If you'd actually want to install a package that is available in multiple repositories without explicitly providing the correct repository name, the package will be installed from the repository with the highest priority.  This priority system is also how the upcoming version 3 of PowerShellGet has been designed to define repository ordering.

Reply:
As of now, whenever a source is opened, it updates the source if it's been more than 5 minutes. `winget source update` is just the explicit command to update.

```
// TODO: Enable some amount of user control over this.
constexpr static auto s_DefaultAutoUpdateTime = 5min;
```